internal/plugin/router_test.go

```diff
@@ -4,20 +4,47 @@ import (
 	"context"
 	"testing"
 
+	"github.com/drone/drone-go/drone"
 	"github.com/drone/drone-go/plugin/converter"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestRouter(t *testing.T) {
+type addNewline struct{}
+
+func newAddNewline() *addNewline {
+	return &addNewline{}
+}
+
+func (p *addNewline) Convert(ctx context.Context, req *converter.Request) (*drone.Config, error) {
+	return &drone.Config{Data: req.Config.Data + "\n"}, nil
+}
+
+func TestConvertRouter(t *testing.T) {
 	t.Parallel()
 
-	r := NewRouter(
-		WithConvertPlugins(
-			NewAddNewline(),
-			NewAddNewline(),
-		),
-	)
-	conf, err := r.Convert(context.Background(), &converter.Request{})
-	assert.NoError(t, err)
-	assert.Equal(t, "\n\n", conf.Data)
+	tests := []struct {
+		router *Router
+		input  string
+		want   string
+	}{
+		// test without convert plugins
```

Comment:
nit: This can be moved to the anonymous struct as a description which can then be included within an assert.  e.g. `assert.Equal(t, test.want, test.got, test.description)`. This makes for easier identifying which table test fails within the logs.